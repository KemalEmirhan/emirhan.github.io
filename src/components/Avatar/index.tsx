import React from 'react';
import Image from 'next/image';

const Avatar = () => {
  return (
    <div className='relative w-16 h-16 sm:w-24 sm:h-24 lg:w-32 lg:h-32 rounded-full border border-gray-200 shadow-sm'>
      <Image
        src='/pp.png'
        alt='Emirhan profile picture'
        fill
        className='object-cover rounded-full'
        priority
      />
    </div>
  );
};

export default Avatar;
