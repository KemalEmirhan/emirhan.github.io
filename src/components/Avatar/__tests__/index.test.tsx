import { render, screen } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import Avatar from '../';

describe('Avatar', () => {
  it('should render', () => {
    const { container } = render(<Avatar />);
    expect(container).toMatchSnapshot();
  });

  it('should render profile image with correct alt text', () => {
    render(<Avatar />);

    const profileImage = screen.getByAltText('Emirhan profile picture');
    expect(profileImage).toBeInTheDocument();
    expect(profileImage).toHaveAttribute(
      'src',
      expect.stringContaining('pp.png')
    );
  });
});
