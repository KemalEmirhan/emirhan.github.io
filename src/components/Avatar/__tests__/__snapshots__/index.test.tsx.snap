// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Avatar > should render 1`] = `
<div>
  <div
    class="relative w-16 h-16 sm:w-24 sm:h-24 lg:w-32 lg:h-32 rounded-full border border-gray-200 shadow-sm"
  >
    <img
      alt="Emirhan profile picture"
      class="object-cover rounded-full"
      data-nimg="fill"
      decoding="async"
      sizes="100vw"
      src="/_next/image?url=%2Fpp.png&w=3840&q=75"
      srcset="/_next/image?url=%2Fpp.png&w=640&q=75 640w, /_next/image?url=%2Fpp.png&w=750&q=75 750w, /_next/image?url=%2Fpp.png&w=828&q=75 828w, /_next/image?url=%2Fpp.png&w=1080&q=75 1080w, /_next/image?url=%2Fpp.png&w=1200&q=75 1200w, /_next/image?url=%2Fpp.png&w=1920&q=75 1920w, /_next/image?url=%2Fpp.png&w=2048&q=75 2048w, /_next/image?url=%2Fpp.png&w=3840&q=75 3840w"
      style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
    />
  </div>
</div>
`;
